docs/content/users/configuration/config.md

```diff
@@ -95,7 +95,13 @@ Composer version for the web container and the [`ddev composer`](../usage/comman
 | :octicons-file-directory-16: project | `2` | Can be `2`, `1`, or empty (`""`) for latest major version at container build time.<br><br>Can also be a minor version like `2.2` for the latest release of that branch, an explicit version like `1.0.22`, or a keyword like `stable`, `preview` or `snapshot`. See Composer documentation.
 
 !!!note "If your `composer.json` requires `composer/composer` that version will be used instead"
-    If your project `composer.json` includes `composer/composer`, then the version specified there will normally be used instead of any version specified by `composer_version`, since `vendor/bin/composer` will come first in the in-container `$PATH`.
+    If your project `composer.json` includes `composer/composer`, then the version specified there will normally be used instead of any version specified by `composer_version`, since `vendor/bin/composer` will come first in the in-container `$PATH`. However, if you prefer to use the version specified by `composer_version`, add the following hook to your `.ddev/config.yaml`:
```

Comment:
This seems like a very unusual workaround. Why would people not want to use the version they specify in their composer.json?

I worry that adding too much detail in places like this just makes it so people can't make it through the docs, or find the right thing when they need it.

Reply:
> Why would people not want to use the version they specify in their composer.json?

It's less about matching the version, and more about issues upgrading composer itself since PHP can't reload classes it's already loaded: https://github.com/composer/composer/discussions/11672

I wonder if this should also be improved upstream. Perhaps composer should never install `vendor/bin/composer`, or if it needs to it should be something like `vendor/bin/composer.internal`.

Reply:
> I worry that adding too much detail in places like this just makes it so people can't make it through the docs, or find the right thing when they need it.

I think if people use Composer in such an advanced way, they'll know how to change something in `$PATH`.
I'll remove this example from the docs because it's unlikely to be found.